### PR TITLE
Dynamic app search

### PR DESCRIPTION
**Pull request by utsengar**

### CODE REVIEW

**Comment by utsengar** [comment]
@adriankonkolowicz Made some minor tweaks and fixed  bug there ".app" was added to UserDefault too, resulting in a bug.

Looks good?

File: HeadsDown/ViewController.swift [hunk 1]
```diff
@@ -51,8 +51,9 @@ class ViewController: NSViewController {
                     let appSelected = result!.lastPathComponent
                     // if path is an application add it to the user preferences
                     if (appSelected != "") {
+                        let appName = String(appSelected.split(separator: ".")[0])
                         var apps = UserPreferences.apps
-                        apps[appSelected] = true
+                        apps[appName] = true
```
Comment (adriankonkolowicz) [inline HeadsDown/ViewController.swift:56]: nice catch!

File: HeadsDown/AppDelegate.swift [hunk 2]
```diff
@@ -27,7 +27,7 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         UserPreferences.isEnabled = true
         DoNotDisturb.isEnabled = false
         
-        //queryApps()
+        NSApp.activate(ignoringOtherApps: true)
```
Comment (adriankonkolowicz) [inline HeadsDown/AppDelegate.swift:36]: What does this do?

**Comment by adriankonkolowicz** [comment]
LGTM, feel free to merge.

File: HeadsDown/AppDelegate.swift (hunk 2 above)
Comment (utsengar) [reply HeadsDown/AppDelegate.swift:36]: Brings window to the forefront when its opened.

**Comment by utsengar** [comment]
One thing I just got a feedback for is we should should hardcode "/Applications" and use `FileManager.SearchPathDirectory.applicationDirectory` https://developer.apple.com/documentation/foundation/filemanager/searchpathdirectory/applicationdirectory.

So, checking how can I open NSOpenPanel with it.

**Comment by adriankonkolowicz** [comment]
Interesting! I wanted to do something like that. 
I tried to do that, setting the `dialog.directoryURL = URL(string: "/Applications")`
Let me check

**Pull request merged**